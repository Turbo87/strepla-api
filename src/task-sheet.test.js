const fs = require('fs');
const path = require('path');
const { parseTaskSheet } = require('./task-sheet');

const FIXTURE_PATH = path.join(__dirname, '..', '__fixtures__', 'tasks');

describe('taskSheet', () => {
  for (let fileName of fs.readdirSync(FIXTURE_PATH)) {
    test(fileName.replace(/\.html$/, ''), () => {
      let body = fs.readFileSync(path.join(FIXTURE_PATH, fileName));
      expect(parseTaskSheet(body)).toMatchSnapshot();
    });
  }
});
