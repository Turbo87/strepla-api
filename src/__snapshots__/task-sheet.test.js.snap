// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`taskSheet ec17-aat-task 1`] = `
Object {
  "turnpoints": Array [
    Object {
      "aatRadial1": null,
      "aatRadial2": null,
      "aatRadius": null,
      "course": null,
      "distance": null,
      "elevation": 114,
      "latitude": 50.847500000000004,
      "longitude": 6.104444444444444,
      "name": "AP2Blausteinsee Mitte",
      "number": "116",
    },
    Object {
      "aatRadial1": 0,
      "aatRadial2": 360,
      "aatRadius": 30000,
      "course": 62,
      "distance": 87300,
      "elevation": 389,
      "latitude": 51.85333333333333,
      "longitude": 7.122777777777777,
      "name": "085Radevormwald Flpl.",
      "number": "085",
    },
    Object {
      "aatRadial1": 0,
      "aatRadial2": 360,
      "aatRadius": 500,
      "course": 244,
      "distance": 93900,
      "elevation": 183,
      "latitude": 50.84722222222222,
      "longitude": 6.102777777777778,
      "name": "055Kronenbrot",
      "number": "055",
    },
    Object {
      "aatRadial1": 78,
      "aatRadial2": 258,
      "aatRadius": 300,
      "course": 168,
      "distance": 1900,
      "elevation": 190,
      "latitude": 50.846944444444446,
      "longitude": 6.103055555555555,
      "name": "ZL-RWY-EDKA",
      "number": "118",
    },
  ],
}
`;

exports[`taskSheet ec17-aat-task2 1`] = `
Object {
  "turnpoints": Array [
    Object {
      "aatRadial1": null,
      "aatRadial2": null,
      "aatRadius": null,
      "course": null,
      "distance": null,
      "elevation": 212,
      "latitude": 50.84666666666667,
      "longitude": 6.103333333333333,
      "name": "086Raststaette ACer-Land",
      "number": "086",
    },
    Object {
      "aatRadial1": 0,
      "aatRadial2": 360,
      "aatRadius": 20000,
      "course": 118,
      "distance": 40300,
      "elevation": 208,
      "latitude": 50.84388888888889,
      "longitude": 6.111944444444444,
      "name": "034Euskirchen BAB S",
      "number": "034",
    },
    Object {
      "aatRadial1": 0,
      "aatRadial2": 360,
      "aatRadius": 8000,
      "course": 312,
      "distance": 41900,
      "elevation": 116,
      "latitude": 50.84805555555556,
      "longitude": 6.104444444444444,
      "name": "004Aldenhoven BAB",
      "number": "004",
    },
    Object {
      "aatRadial1": 0,
      "aatRadial2": 360,
      "aatRadius": 20000,
      "course": 58,
      "distance": 64700,
      "elevation": 236,
      "latitude": 51.85333333333333,
      "longitude": 7.1175,
      "name": "098Solingen Kreisfriedhof",
      "number": "098",
    },
    Object {
      "aatRadial1": 0,
      "aatRadial2": 360,
      "aatRadius": 500,
      "course": 237,
      "distance": 73900,
      "elevation": 183,
      "latitude": 50.84722222222222,
      "longitude": 6.102777777777778,
      "name": "055Kronenbrot",
      "number": "055",
    },
    Object {
      "aatRadial1": null,
      "aatRadial2": null,
      "aatRadius": null,
      "course": 168,
      "distance": 1900,
      "elevation": 190,
      "latitude": 50.846944444444446,
      "longitude": 6.103055555555555,
      "name": "ZL-RWY-EDKA",
      "number": "118",
    },
  ],
}
`;

exports[`taskSheet ec17-racing-task 1`] = `
Object {
  "turnpoints": Array [
    Object {
      "aatRadial1": null,
      "aatRadial2": null,
      "aatRadius": null,
      "course": null,
      "distance": null,
      "elevation": 153,
      "latitude": 50.846111111111114,
      "longitude": 6.101666666666667,
      "name": "002Aachen Europaplatz",
      "number": "002",
    },
    Object {
      "aatRadial1": null,
      "aatRadial2": null,
      "aatRadius": null,
      "course": 117,
      "distance": 91000,
      "elevation": 292,
      "latitude": 50.84,
      "longitude": 7.120833333333333,
      "name": "062Maria Laach Abtei",
      "number": "062",
    },
    Object {
      "aatRadial1": null,
      "aatRadial2": null,
      "aatRadius": null,
      "course": 236,
      "distance": 78800,
      "elevation": 424,
      "latitude": 49.83305555555556,
      "longitude": 6.105555555555555,
      "name": "103Utscheid",
      "number": "103",
    },
    Object {
      "aatRadial1": null,
      "aatRadial2": null,
      "aatRadius": null,
      "course": 33,
      "distance": 70000,
      "elevation": 327,
      "latitude": 50.841944444444444,
      "longitude": 6.1147222222222215,
      "name": "031Effelsberg Radiot.",
      "number": "031",
    },
    Object {
      "aatRadial1": null,
      "aatRadial2": null,
      "aatRadius": null,
      "course": 306,
      "distance": 60600,
      "elevation": 183,
      "latitude": 50.84722222222222,
      "longitude": 6.102777777777778,
      "name": "055Kronenbrot",
      "number": "055",
    },
    Object {
      "aatRadial1": null,
      "aatRadial2": null,
      "aatRadius": null,
      "course": 168,
      "distance": 1900,
      "elevation": 190,
      "latitude": 50.846944444444446,
      "longitude": 6.103055555555555,
      "name": "ZL-RWY-EDKA",
      "number": "118",
    },
  ],
}
`;
